roadmap/implementers-guide/src/runtime-api/README.md

```diff
@@ -0,0 +1,172 @@
+	availability: Bitfield,
+}
+
+struct ScheduledCore {
+	/// The ID of a para scheduled.
+	para: ParaId,
+	/// The collator required to author the block, if any.
+	collator: Option<CollatorId>,
+}
+
+enum CoreState {
+	/// The core is currently occupied.
+	Occupied(OccupiedCore),
+	/// The core is currently free, with a para scheduled and given the opportunity
+	/// to occupy.
+	///
+	/// If a particular Collator is required to author this block, that is also present in this
+	/// variant.
+	Scheduled(ScheduledCore),
+	/// The core is currently free and there is nothing scheduled. This can be the case for parathread
+	/// cores when there are no parathread blocks queued. Parachain cores will never be left idle.
+	Free,
+}
+```
+
+## Global Validation Schedule
+
+Yields the [`GlobalValidationSchedule`](../types/candidate.md#globalvalidationschedule) at the state of a given block. This applies to all para candidates with the relay-parent equal to that block.
+
+```rust
+fn global_validation_schedule() -> GlobalValidationSchedule;
+```
+
+## Local Validation Data
+
+Yields the [`LocalValidationData`](../types/candidate.md#localvalidationdata) for the given [`ParaId`](../types/candidate.md#paraid) along with an assumption that should be used if the para currently occupies a core: whether the candidate occupying that core should be assumed to have been made available and included or timed out and discarded, along with a third option to assert that the core was not occupied. This choice affects everything from the parent head-data, the validation code, and the state of message-queues. Typically, users will take the assumption that either the core was free or that the occupying candidate was included, as timeouts are expected only in adversarial circumstances and even so, only in a small minority of blocks directly following validator set rotations.
+
+The documentation of [`LocalValidationData`](../types/candidate.md#localvalidationdata) has more information on this dichotomy.
+
+```rust
+/// An assumption being made about the state of an occupied core.
+enum OccupiedCoreAssumption {
+	/// The candidate occupying the core was made available and included to free the core.
+	Included,
+	/// The candidate occupying the core timed out and freed the core without advancing the para.
+	TimedOut,
+	/// The core was not occupied to begin with.
+	Free,
+}
+
+/// Returns the local validation data for the given para and occupied core assumption.
+///
+/// Returns `None` if either the para is not registered or the assumption is `Freed`
+/// and the para already occupies a core.
+fn local_validation_data(ParaId, OccupiedCoreAssumption) -> Option<LocalValidationData>;
+```
+
+## Session Index
+
+Get the session index that is expected at the child of a block.
+
+In the [`Initializer`](../runtime/initializer.md) module, session changes are buffered by one block. The session index of the child of any block is always predictable by that block's state.
```

Comment:
```suggestion
In the [`Initializer`](../runtime/initializer.md) module, session changes are buffered by one block. The session index of the child of any relay block is always predictable by that block's state.
```

I seem to remember discussion of the idea that sessions change between blocks, and if a parachain doesn't issue new blocks for every relay chain block, it is possible for the parachain to guess wrongly which session its new block is in if it looks only at the parachain parent.